scripts/OverlaySurvey.py

```diff
@@ -0,0 +1,194 @@
+# importing the requests library 
+import requests 
+import json
+import time
+import sys
+import os
+import argparse
+import networkx as nx
+from networkx import Graph
+from collections import defaultdict
+
+def add_new_node(label):
+    if G.has_node(label):
+        return
+    G.add_node(label, label=label)
+
+def add_new_edge(u, v):
+    if G.has_edge(u, v):
+        return
+    G.add_edge(u, v)
+
+def get_next_peers(topology):
+    results = []
+    for key in topology:
+        if topology[key] is None:
+                continue
+
+        if "inboundPeers" in topology[key] and topology[key]["inboundPeers"]:
+            for in_peers in topology[key]["inboundPeers"]:
+                results.append(in_peers["nodeId"])
+        if "outboundPeers" in topology[key] and topology[key]["outboundPeers"]:
+            for out_peers in topology[key]["outboundPeers"]:
+                results.append(out_peers["nodeId"])
+    return results
+
+def send_requests(peer_list):
+    for key in peer_list:
+        PARAMS["node"] = key
+        requests.get(url = SURVEY_REQUEST, params = PARAMS) 
+
+def check_results():
+    r = requests.get(url = SURVEY_RESULT)
+    data = r.json()
+
+    if("topology" not in data):
+        return []
+
+    topology = data["topology"]
+
+    for key in topology:
+        if topology[key] is None:
+            continue
+
+        if "inboundPeers" in topology[key] and topology[key]["inboundPeers"]:
```

Comment:
This could read easier if you made some temporaries, eg. `curr = topology[key]`, `merged = merged_results[key]`, and `other_key = in_peers["nodeId"]` and `other_key = out_peers["nodeId"]` in each loop.

Reply:
Beyond that, this is almost identical code appearing twice. Let's review it just once by refactoring it into a function.

Reply:
done